boutdata/tests/make_test_data.py

```diff
@@ -0,0 +1,390 @@
+            var.setncattr(key, value)
+        var.setncattr("yindex_global", fieldperp_global_yind)
+
+        result[name] = data[xslice, zslice]
+
+    createPerp("fieldperp_1")
+    createPerp("fieldperp_2")
+
+    # Time-dependent array
+    def createScalar_t(name):
+        var = outputfile.createVariable(name, float, ("t",))
+
+        data = rng.random(nt)
+        var[:] = data
+
+        result[name] = data
+
+    createScalar_t("t_array")
+    createScalar_t("scalar_t_1")
+    createScalar_t("scalar_t_2")
+
+    # Scalar
+    def createScalar(name, value):
+        var = outputfile.createVariable(name, type(value))
+
+        var[...] = value
+
+        result[name] = value
+
+    createScalar("BOUT_VERSION", 4.31)
+    for key, value in grid_info.items():
+        createScalar(key, value)
+    nxpe = grid_info["NXPE"]
+    createScalar("PE_XIND", i % nxpe)
+    createScalar("PE_YIND", i // nxpe)
+    createScalar("MYPE", i)
+
+    return result
+
+
+def concatenate_data(data_list, *, nxpe, fieldperp_yproc_ind):
+    # Just keep scalars from root file
+    result = copy(data_list[0])
+    for x in list(result.keys()):
+        if x[:5] == "field":
+            result.pop(x)
+
+    npes = len(data_list)
+    nype = npes // nxpe
+    if npes % nxpe != 0:
+        raise ValueError("nxpe=%i does not divide len(data_list)=%i".format(nxpe, npes))
+
+    for var in ("field3d_t_1", "field3d_t_2", "field2d_t_1", "field2d_t_2"):
+        # Join in x-direction
+        parts = [
+            np.concatenate(
+                [data_list[j][var] for j in range(i * nxpe, (i + 1) * nxpe)], axis=1
+            )
+            for i in range(nype)
+        ]
+        # Join in y-direction
+        result[var] = np.concatenate(parts, axis=2)
+
+    for var in ("field3d_1", "field3d_2", "field2d_1", "field2d_2"):
+        # Join in x-direction
+        parts = [
+            np.concatenate(
+                [data_list[j][var] for j in range(i * nxpe, (i + 1) * nxpe)], axis=0
+            )
+            for i in range(nype)
+        ]
+        # Join in y-direction
+        result[var] = np.concatenate(parts, axis=1)
+
+    for var in ("fieldperp_t_1", "fieldperp_t_2"):
+        # Join in x-direction
+        result[var] = np.concatenate(
+            [
+                data_list[j][var]
+                for j in range(
+                    fieldperp_yproc_ind * nxpe, (fieldperp_yproc_ind + 1) * nxpe
+                )
+            ],
+            axis=1,
+        )
+
+    for var in ("fieldperp_1", "fieldperp_2"):
+        # Join in x-direction
+        result[var] = np.concatenate(
+            [
+                data_list[j][var]
+                for j in range(
+                    fieldperp_yproc_ind * nxpe, (fieldperp_yproc_ind + 1) * nxpe
+                )
+            ],
+            axis=0,
+        )
+
+    return result
+
+
+def apply_slices(expected, tslice, xslice, yslice, zslice):
```

Comment:
If you'd time to write a short docstring here as well, that would be great. Can for example `tslice` be a `slice` object?